Device Type for SH20T,SH25T and hex output of device code for unknown devices

ptar opened the pull request:
* Device Type for SH20T is 0x0E26
* Output device code for unknown device types to ease adding new devices types

fstierle commented:
I am currently testing on a SH25T, the Code there is 0x0E28, can you add this?

ptar commented:
Thank you for the information!
I added it.

ptar commented:
> I am currently testing on a SH25T

Could you please check, if the changes to get charging and discharging power described in https://github.com/mkaiser/Sungrow-SHx-Inverter-Modbus-Home-Assistant/discussions/310#discussioncomment-10054375 are working for your SH25T, too?

pep1 commented:
Hi @ptar thanks for adding the device types. I wonder whether it's worth adding the entire list of new devices in one go. The codes are documented [here](https://github.com/Gnarfoz/Sungrow-Inverter/blob/main/Modbus%20Information/TI_20231019_Communication.Protocol.of.Residential.Hybrid.Inverter_V1.1.2_EN.pdf).

![Screenshot from 2024-07-16 17-13-04](https://github.com/user-attachments/assets/e1b55c76-6d7b-40a8-88af-a79bc4e06362)

Note that the codes need an additional `0` to be parsed correctly, e.g. `0x0E26` instead of `0xE26` like stated in the doc.

Also it looks like there is a convention to include the model version in the device type, like this: `SH{15|20|25}-V11`

fstierle commented:
I added all of these and made an extra pull request.

@ptar we currently have no battery on the inverter we are waiting for the battery series is available to us

ptar commented:
> I added all of these and made an extra pull request.

Fine!

I had added the device type code in hex to the last line for unknown device codes:
`            Unknown device code: {{ '%0x' % (states('sensor.sungrow_device_type_code') | int(default=0))  }}`

This made it easier for me to get the code without having the documentation.
Maybe you could add it to your PR, too, to help others with upcoming codes?

mkaiser commented:
did not test, but looks good, thank you! :)
